### PR TITLE
Feature: add .nomedia to directories with checkbox in Properties

### LINKED ISSUE #769: [FR] Handle `.nomedia`
Hi! Good old CyanogenMod FM had this feature: in a folder's properties there was the checkbox that created/deleted `.nomedia` file.
To see it working here is the apk: https://f-droid.org/archive/com.cyanogenmod.filemanager.ics_1015.apk

### PR DESCRIPTION
**Pull request by bowiechen**
Fixes #769, referenced #844.

Steps to reproduce:
1. Tap the "three dot" button the right of a folder.
2. Tap "Properties".
3. A checkbox labeled "Hide from media browsers" should be visible and its state reflecting the existence of the `.nomedia` file in that directory.
4. Toggle the checkbox and tap "OK". If the `.nomedia` file exists, it will be deleted; if the `.nomedia` file does not exist, it will be created.

Toggling the checkbox will not immediately delete the `.nomedia` file -- "OK" must be tapped in order to commit the action.

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/ui/dialogs/GeneralDialogCreation.java [hunk 1]
```diff
@@ -380,12 +380,15 @@ private static void showPropertiesDialog(final HybridFileParcelable baseFile, fi
                 name = baseFile.getName(),
                 parent = baseFile.getReadablePath(baseFile.getParent(c));
 
+        File nomediaFile = baseFile.isDirectory()? new File(baseFile.getPath() + "/" + FileUtils.NOMEDIA_FILE) : null;
```
[the remark this answers is not in the document]
Comment (bowiechen) [reply app/src/main/java/com/amaze/filemanager/ui/dialogs/GeneralDialogCreation.java]: Are you concerned if a user-created `.nomedia` will be destroyed with this line? If so, I don't think we need to worry about this: users who are tech-savvy enough know what `.nomedia` is and those who aren't will not name one of their files `.nomedia`.

[the remark this answers is not in the document]
Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/ui/dialogs/GeneralDialogCreation.java]: > Are you concerned if a user-created .nomedia will be destroyed with this line? 

User created nomedia being destroyed when someone unticks the box is a feature, that's what unticking the box does.

[the remark this answers is not in the document]
Comment (bowiechen) [reply app/src/main/java/com/amaze/filemanager/ui/dialogs/GeneralDialogCreation.java]: Ah I see what you mean. If the folder is a SMB folder, for example, it'll probably most likely break. We should limit this to only local folders. 

Good catch, thanks!

**Comment by VishalNehra** [comment]
Code changes seem fine, only thing, the checkbox shouldn't be placed in properties dialog. The dialog is already pretty big so much that on small devices we have to scroll it. A dialog with scroll won't probably look good. 
I suggest you place the option in three dot menu on top right, after the folders are selected. That way we'll have one more feature to create .nomedia in a single operation.

**Comment by EmmanuelMess** [comment]
> I suggest you place the option in three dot menu on top right, after the folders are selected. That way we'll have one more feature to create .nomedia in a single operation.

I think there's not enough space in the selection contextual toolbar already. Also, the tick box will be easily missed there.

**Comment by bowiechen** [comment]
> there is not enough space in the selection contextual toolbar

I think what @VishalNehra was referring to is to add a checkbox in the overflow menu. A checkbox in the overflow menu could be a good solution, however, I personally think adding to the overflow context toolbar will confuse users with this feature and the other "hidden files" function, which is purely cosmetic and doesn't affect the filesystem. I vote to keep the checkbox in Properties for this reason.

**Pull request merged**